### PR TITLE
ci: adding testing with mac M1

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

reproducing #2383
ref: https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2385.org.readthedocs.build/en/2385/

### CODE REVIEW

**Comment by Borda** [comment]
@justusschock could you pls help here to check why the metrics are off?

**Comment by Borda** [comment]
full run in https://github.com/Lightning-AI/torchmetrics/actions/runs/8079049135

File: tests/unittests/audio/test_pesq.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def test_on_real_audio():
     """Test that metric works as expected on real audio signals."""
     rate, ref = wavfile.read(_SAMPLE_AUDIO_SPEECH)
     rate, deg = wavfile.read(_SAMPLE_AUDIO_SPEECH_BAB_DB)
-    pesq = perceptual_evaluation_speech_quality(torch.from_numpy(deg), torch.from_numpy(ref), rate, "wb")
-    assert pesq == 1.0832337141036987
-    pesq = perceptual_evaluation_speech_quality(torch.from_numpy(deg), torch.from_numpy(ref), rate, "nb")
-    assert pesq == 1.6072081327438354
+    pesq_score = perceptual_evaluation_speech_quality(torch.from_numpy(deg), torch.from_numpy(ref), rate, "wb")
+    assert torch.round(pesq_score, decimals=4) == 1.0832
```
Comment (justusschock) [inline tests/unittests/audio/test_pesq.py:134]: can we still use allclose here? Depending on the actual number, they sometimes cannot even be displayed as a floating point but only a very close approximation.

File: tests/unittests/audio/test_pesq.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def test_on_real_audio():
...
+    pesq_score = perceptual_evaluation_speech_quality(torch.from_numpy(deg), torch.from_numpy(ref), rate, "nb")
+    assert torch.round(pesq_score, decimals=4) == 1.6072
```
Comment (justusschock) [inline tests/unittests/audio/test_pesq.py:136]: same as above

**Review by justusschock — APPROVED** [review]

File: tests/unittests/audio/test_c_si_snr.py [hunk 3]
```diff
@@ -73,9 +73,9 @@ def test_on_real_audio():
     deg_stft = torch.stft(deg, n_fft=256, hop_length=128, return_complex=True)
 
     v = complex_scale_invariant_signal_noise_ratio(deg_stft, ref_stft, zero_mean=False)
-    assert torch.allclose(v, torch.tensor(0.03019072115421295, dtype=v.dtype), atol=1e-4), v
+    assert torch.allclose(v, torch.tensor(0.03019, dtype=v.dtype), atol=1e-4), v
```
Comment (Borda) [inline tests/unittests/audio/test_c_si_snr.py:76]:
```suggestion
    assert torch.allclose(v, torch.tensor(0.03019072115421295, dtype=v.dtype), atol=1e-4), v
```

File: tests/unittests/audio/test_c_si_snr.py [hunk 4]
```diff
@@ -73,9 +73,9 @@ def test_on_real_audio():
...
     v = complex_scale_invariant_signal_noise_ratio(deg_stft, ref_stft, zero_mean=True)
-    assert torch.allclose(v, torch.tensor(0.030391741544008255, dtype=v.dtype), atol=1e-4), v
+    assert torch.allclose(v, torch.tensor(0.03039, dtype=v.dtype), atol=1e-4), v
```
Comment (Borda) [inline tests/unittests/audio/test_c_si_snr.py:78]:
```suggestion
    assert torch.allclose(v, torch.tensor(0.030391741544008255, dtype=v.dtype), atol=1e-4), v
```

File: tests/unittests/audio/test_stoi.py [hunk 5]
```diff
@@ -126,7 +126,7 @@ def test_on_real_audio():
     rate, deg = wavfile.read(_SAMPLE_AUDIO_SPEECH_BAB_DB)
     assert torch.allclose(
         short_time_objective_intelligibility(torch.from_numpy(deg), torch.from_numpy(ref), rate).float(),
-        torch.tensor(0.6739177),
+        torch.tensor(0.67391),
```
Comment (Borda) [inline tests/unittests/audio/test_stoi.py:129]:
```suggestion
        torch.tensor(0.6739177),
```

File: tests/unittests/audio/test_pesq.py (hunk 1 above)
Comment (Borda) [reply tests/unittests/audio/test_pesq.py]: done, good point :)

**Pull request merged**